module/VuFind/src/VuFind/Db/Migration/MigrationLoader.php

```diff
@@ -85,16 +87,27 @@ public function getMigrationSubdirectoriesMatchingVersion(string $oldVersion, st
         }
 
         // Find version-appropriate subdirectories in the migration directory:
-        return array_values(
-            array_filter(
-                glob("$basePath/*"),
-                function ($path) use ($oldVersion) {
-                    $parts = explode('/', $path);
-                    $version = array_pop($parts);
-                    return preg_match('/^\d/', $version) && Comparator::greaterThanOrEqualTo($version, $oldVersion);
-                }
-            )
+        $directories = array_filter(
+            glob("$basePath/*"),
+            function ($path) use ($oldVersion) {
+                $parts = explode(DIRECTORY_SEPARATOR, $path);
+                $version = array_pop($parts);
+                return preg_match('/^\d/', $version) && Comparator::greaterThanOrEqualTo($version, $oldVersion);
+            }
+        );
+
+        usort(
+            $directories,
+            function ($a, $b): int {
+                $aParts = explode(DIRECTORY_SEPARATOR, $a);
+                $bParts = explode(DIRECTORY_SEPARATOR, $b);
+                $aVersion = end($aParts);
+                $bVersion = end($bParts);
+                return $aVersion === $bVersion ? 0 : (Comparator::greaterThan($aVersion, $bVersion) ? 1 : -1);
```

Comment:
It occurs to me that this might be an even better version, just in case the order of the data is not as we expect to begin with:

```suggestion
                return $aVersion === $bVersion
                    ? $a <=> $b
                    : (Comparator::greaterThan($aVersion, $bVersion) ? 1 : -1);
```

Reply:
I suppose there's no problem doing this, but the method itself must somehow fail for this to make a difference. :)

Reply:
Just thinking ahead to a possible future where we change the glob call for some reason and forget to change this bit. ;-)